### PR TITLE
Add metric "keboola.go.templates.repository.update.duration"

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-208

Changes:
- Added metric `keboola.go.templates.repository.update.duration`.

### CODE REVIEW

File: internal/pkg/telemetry/fortest.go [hunk 1]
```diff
@@ -34,32 +33,55 @@ type ForTest interface {
...
+}
+
+// WithDataPointSortKey set a function to generate sort key for each data point.
+// DataPoints are internally represented as a map, so they have random order.
+func WithDataPointSortKey(v func(attrs attribute.Set) string) TestMeterOption {
+	return func(cnf *assertMetricConfig) {
+		cnf.dataPointSortKey = v
+	}
+}
+
```
Comment (michaljurecko) [inline internal/pkg/telemetry/fortest.go:84]: Pridana moznost mapovat v test attrs nie len pre span, ale aj pre meter + nastavenie zoradzovania data points v metrike.

File: internal/pkg/service/common/dependencies/dependencies.go [hunk 2]
```diff
@@ -112,6 +112,7 @@ type Mocked interface {
 	Project
 
 	DebugLogger() log.DebugLogger
+	TestTelemetry() telemetry.ForTest
```
Comment (michaljurecko) [inline internal/pkg/service/common/dependencies/dependencies.go:115]: V interface mi chybala tato metoda.

File: internal/pkg/template/repository/manager/manager.go [hunk 3]
```diff
@@ -59,10 +62,20 @@ type dependencies interface {
 }
 
 func New(ctx context.Context, d dependencies, defaultRepositories []model.TemplateRepository) (*Manager, error) {
+	meter, err := d.Telemetry().Meter().Float64Histogram(
+		"keboola.go.templates.repository.update.duration",
+		metric.WithDescription("Templates repository update duration."),
+		metric.WithUnit("ms"),
+	)
```
Comment (michaljurecko) [inline internal/pkg/template/repository/manager/manager.go:69]: Templates API si kazdych 5 min kontroluje Git repositoyr so sablonami, ci nastala nejaka zmena.

File: internal/pkg/template/repository/manager/manager.go [hunk 4]
```diff
@@ -130,19 +143,34 @@ func (m *Manager) Update(ctx context.Context) <-chan error {
...
+			elapsedTime := float64(time.Since(startTime)) / float64(time.Millisecond)
+
+			// Metric
+			m.updateMeter.Record(ctx, elapsedTime, metric.WithAttributes(
+				attribute.String("repo_name", repoDef.Name),
+				attribute.String("repo_url", repoDef.URL),
+				attribute.String("repo_ref", repoDef.Ref),
+				attribute.Bool("is_success", err == nil),
+				attribute.Bool("is_changed", changed),
+			))
```
Comment (michaljurecko) [inline internal/pkg/template/repository/manager/manager.go:164]: Atributy novej metriky.

File: internal/pkg/template/repository/manager/manager_test.go [hunk 5]
```diff
@@ -82,6 +90,90 @@ func TestRepository(t *testing.T) {
...
+							Bounds: histBounds,
+							Attributes: attribute.NewSet(
+								attribute.String("repo_name", "keboola"),
+								attribute.String("repo_url", "file://<tmp_dir>"),
+								attribute.String("repo_ref", "main"),
+								attribute.Bool("is_success", true),
+								attribute.Bool("is_changed", true),
+							),
+						},
+					},
```
Comment (michaljurecko) [inline internal/pkg/template/repository/manager/manager_test.go:160]: Unit test, 1x ziadna zmena, 2x zmena.

Testuje sa to oproti Git tmpl directory `file://....`.

**Review by zajca — APPROVED** [review]

**Pull request merged**